scanomatic/generics/abstract_model_factory.py

```diff
@@ -1306,8 +1299,8 @@ def unserialize(serialized_obj: Union[str, bool, Generator], dtype):
 
         else:
             try:
-                return pickle.loads(serialized_obj)
-            except (pickle.PickleError, TypeError):
+                return pickle.loads(serialized_obj.encode())
+            except (pickle.PickleError, TypeError, EOFError):
```

Comment:
I think the `encode()` is fine, but maybe not the `EOFError`..?

Reply:
It might be fine, but perhaps logging exception so we at least can track the failings.

Reply:
The `EOFError` is an indication that something went wrong earlier in the chain (the symptom is a zero length byte-array I think), but so are the other errors, so it is not unique in that respect. I'll add logging.